File: src/test/java/com/circleescape/server/persistence/ScoreRepositoryIntegrationTest.java
package com.circleescape.server.persistence;

import com.circleescape.server.scoreboard.data.ScoreRepository;
import io.restassured.RestAssured;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.http.HttpStatus;

import java.util.Map;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;

@SpringBootTest(webEnvironment = RANDOM_PORT)
public class ScoreRepositoryIntegrationTest {
    @Autowired
    private ScoreRepository nameRepository;

    @Value("${local.server.port}")
    private int serverPort;

    @Test
    public void triggerTest() {

        //assert data is even loaded
        assertEquals(2, nameRepository.count());
        System.out.println("port: "+serverPort);

        RestAssured
                .given()
                .baseUri("http://localhost")
                .port(serverPort)
                .basePath("game")
                .body("Jack")

                .when()
                .post("/4/postName")

                .then()
                .statusCode(HttpStatus.OK.value());

        //assert new name is added
        assertEquals(3, nameRepository.count());
        assertEquals("Jack", nameRepository.findById(3L).get().getName());
    }
}

File: src/test/java/com/circleescape/server/persistence/ScoreRepositoryIntegrationTest.kt
package com.circleescape.server.persistence

import org.springframework.boot.test.context.SpringBootTest
import org.springframework.boot.test.context.SpringBootTest.WebEnvironment
import org.springframework.beans.factory.annotation.Autowired
import com.circleescape.server.scoreboard.data.ScoreRepository
import io.restassured.RestAssured
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Value
import org.springframework.http.HttpStatus

@SpringBootTest(webEnvironment = WebEnvironment.RANDOM_PORT)
class ScoreRepositoryIntegrationTest {
    @Autowired
    private val nameRepository: ScoreRepository? = null

    @Value("\${local.server.port}")
    private val serverPort = 0
    @Test
    fun triggerTest() {

        //assert data is even loaded
        Assertions.assertEquals(2, nameRepository!!.count())
        println("port: $serverPort")
        RestAssured
            .given()
            .baseUri("http://localhost")
            .port(serverPort)
            .basePath("scoreboard")
            .body("Jack")
            .`when`()
            .post("/4/postName")
            .then()
            .statusCode(HttpStatus.OK.value())

        //assert new name is added
        Assertions.assertEquals(3, nameRepository.count())
        Assertions.assertEquals("Jack", nameRepository.findById(3L).get().name)
    }
}